app/components/WebsocketProvider.tsx

```diff
@@ -413,56 +404,40 @@ class WebsocketProvider extends React.Component<Props> {
       }
     );
 
-    // received when a user is given access to a collection
-    // if the user is us then we go ahead and load the collection from API.
+    this.socket.on("collections.add_user", async (event: Membership) => {
+      memberships.add(event);
+      await collections.fetch(event.collectionId, {
```

Comment:
Earlier, collection was fetched _only_ for the user who was added, not it'll be fetched for all users in the collection room?

Reply:
Ah yep, thanks – I made this not a forced fetch for a bit in which case it didn't matter, but then moved back to a force fetfh

Reply:
Also, since `collections.add_user` endpoint is used not just for adding a user, but also for _updating_ an existing user's permission, then, we aren't updating the document policies for that case as we're doing earlier? I don't also see any hooks particularly for achieving that 👀

Reply:
It's true, thinking of using a lifecycle hook on membership update – thoughts?

Reply:
If we can achieve a _full cascade_, why not? I mean, since for this particular instance, a membership update should affect two other corresponding model instances:
1. Respective collection instance(w.r.t policies)
2. All documents within that collection(again w.r.t policies)

If we can achieve cascade from `membership` update -> `collection` policy update -> `documents'` policy update, we should be good?

Hope I was clear here 🤞

Reply:
<img width="599" alt="image" src="https://github.com/user-attachments/assets/d2f9b907-2ca8-4159-9c1e-65942f4826f7">

Reply:
Ah yea, I have this on Membership but alternatively a similar hook could be on policy, perhaps that is better as it's more _direct_.

Reply:
Added here:
https://github.com/outline/outline/pull/7460/files#diff-66b70c39db74930a46b4a70e8c8c543bcc7881808c5b8ee1c1c7f8107ff9f587R33

Reply:
The problem is there is no relationship between policies, you'd need to go via the parent model. Definitely doable, but getting quite complex~ possible followup